src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -380,7 +380,7 @@ public virtual void Cancel(bool force)
 
             if (!tThread.Join(0))
             {
-                log.Debug("Killing thread {0} for cancel", tThread.ManagedThreadId);
+                log.Debug("Killing thread {0} for cancel", thread.ManagedThreadId);
```

Comment:
Yes... a number of things are out of whack with this commit... see my reply to @rprouse comment.
I'm going through it now to straighten it out.